\documentclass[letterpaper]{article}
\usepackage{proceed2e}
\usepackage[margin=1in]{geometry}
\usepackage{times}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}

\usepackage{tikz}
\usetikzlibrary{calc}
\newcommand{\tikzmark}[1]{\tikz[overlay,remember picture] \node (#1) {};}
\newcommand{\underarrow}[2] {
  \begin{tikzpicture}[overlay,remember picture,out=340,in=210,distance=0.3cm]
    \draw [->,shorten >=3pt,shorten <=-3pt] ({#1}.south) to ({#2}.west);
  \end{tikzpicture}
}  
\newcommand{\Ex}{\mathbb{E}}
\usepackage{graphicx}
\usepackage{dblfloatfix}
%\usepackage[superscript,biblabel]{cite}
\usepackage{multirow,tabularx}
\usepackage{hhline}
\usepackage{amsmath}
\usepackage{cancel}

\usepackage[style=nature,
  citestyle=authoryear,
  maxcitenames=3,
  backend=biber,
  date=year]{biblatex}
%\bibliographystyle{apa}
\addbibresource{uiai.bib}

\newcommand\indep{\protect\mathpalette{\protect\independenT}{\perp}}
\def\independenT#1#2{\mathrel{\rlap{$#1#2$}\mkern2mu{#1#2}}}

\title{Inferring Causality from Finite Data using Conditional Independence}
%\author{Daniel Speyer\\Columbia University}
\author{}
\begin{document}

\maketitle

\begin{abstract}
  We propose techniques for inferring causal structure from limited
  observational data.  These techniques produce less extensive results
  than those which require exact joint probabilities, but are more
  practical.  The general principle is to describe multiple
  possible causal models in sufficient detail to have with specific
  consequences and test those
  consequences.  After developing these techniques abstractly, we test
  them in simulation and on real world Crohn's Disease data.
\end{abstract}

\section{Introduction}
Inferring causal graphs from observational data is a widely-sought
goal in statistics.  It is especially important in medicine, where
finding the cause of a disease may provide a way to cure or prevent
it, but finding an effect of a disease is of little practical value.
Medicine is also a context in which observational data is plentiful and finding
correlations is easy, but running intervention studies is slow,
expensive, regulated, and potentially dangerous.

The most common tool for causal inference is conditional
independence.  The specifics of a causal graph determine which node
are independent conditioned on which others by the ``bayes ball''
rule, and therefore it should be possible to observe the independences
and work backward to the causal graph.

This has proved to be more difficult in practice.  In particular,
observing independence is not as simple as it sounds.  [\cite{Pearl}] suggests
we test joint conditional probability distributions for equality.
Leaving aside the curse of dimensionality, this assumes we have the
exact distributions.  If all we have is a finite random sample from
those distributions, we can construct posteriors, but we cannot perform
an equality test.  [\cite{Spirtes}] encourage us to assume that
``the statistical decisions required by the [causal inference]
algorithms are correct
for the population'', while admitting that this requirement ``is often not met in
practice''.  Others have described this as ``possessing an
independence oracle'' [\cite{Peters,pcalg}].  Many simply include a
``test if $a\indep b|s$'' step in their algorithms, assuming the
reader will already know how.

In practice, the usual solution [\cite{akelleh,pcalg}] is to treat independence as a null
hypothesis, try to reject it at some p threshold, and treat any
failure as establishing it.  Needless to say, this is incorrect.
[\cite{bayesnet}] punts the responsibility to the caller instead,
which, while not actually wrong, is unhelpful

Dealing with finite data means the possibility of dealing with too
little data.  The elegant solution is to give some numerical
expression of confidence which becomes ``I don't know'' when the data
gets too small.  This solution has another benefit: in the biomedical
context, it is routine to test thousands of equally plausible
hypotheses at once.  A flat accuracy of 99\% is unhelpful in the face
of this, but a numeric expression of confidence allows proper
compensation.

Traditional inference algorithms such as [\cite{Pearl}] generally
construct the entire DAG by a series of logical deductions that depend
on each other.  Such approaches do not translate easily to
uncertainty.  If we took the posterior probability for each
successive test in a traditional algorithm and multiplied them to get
a posterior for the entire graph, the resulting posterior would likely
be too low for any use, and might not be the overall plurality.
Furthermore, the number of possible causal graphs is roughly
exponential in the number of variables, so finding probabilities for
all of them is intractable outside of toy problems (even
\textit{storing} those probabilities is beyond the reach of real-world
hardware for most interesting problems in medicine!).

It may be possible to usefully approximate entire-DAG inference, but
we will attempt a less ambitious solution: to infer, as accurately as
possible, the direction of a single arrow.  Specifically, one pointing
from a variable we can easily intervene on (a ``cure'') to one we care
about (a ``disease'').

\subsection{Motivating Problem: Crohn's Disease}

This paper is optimized around a specific practical problem:
untangling the microbiome's role in Illial Crohn's Disease.  It is
well established that there are many differences in the intestinal
bacteria of healthy people and of people with the disease
[\cite{hofer}], but it is not established which (if any) of the differences of
bacteria \textit{cause} the disease.  
There have been attempts to determine this by randomized controlled
trial, but early results are discouraging [\cite{rctma}] and the number
of species, combined with other relevant
variables, make exhaustive RCTs impractical.

Data is available on this problem from [\cite{data}], including
genetic, microbiome and
health information, but for only 58 patients.  The microbiome
information is a series of 16S reads, but can generally be described
as ``present'' or ``absent'', with very low concentrations of a
species rounded off to ``absent''.  This comes much closer to fitting
the empirical distributions than any simple scalar formula, which is
why this paper uses binary variables.

\section{Extending Graphs}

Let us begin with the simplest case.  Suppose we have a known cause, a
known effect, and a variable which connects to the effect in an
unknown way.   For Crohn's Disease, the known cause is mutations in the NOD2
gene [\cite{nod2}], the known effect is the disease, and the unknown
is each of 222 species of plausibly relevant bacteria, taken
independently.  To help keep our example in mind, we will refer to
these factors as $G$ (``gene''), $D$ (``disease'') and $B$
(``bacterium'').  So our initial graph is $G \rightarrow D$ -- $B$.
We do not observe a
correlation between $G$ and $B$, but that might only mean our test is
underpowered.  

For now, we will assume that there is no direct causal
link between $G$ and $B$, and furthermore that there are no unobserved
confounders or selection effects.  We will consider these later.
This leaves us only two models, $G \rightarrow D \rightarrow B$ or $G
\rightarrow D \leftarrow B$.  We can call these ``chain'' and
``collide'' models, or $m_{ch}$ and $m_{co}$ for short.  Can we
distinguish between them?

Yes.  Let us consider
$p(G,B)$:

\begin{eqnarray*}
p(G,B|m_{ch}) & = & \sum_D p(G)p(D|G)p(B|D) \\
p(G,B|m_{co}) & = & p(G)p(B)
\end{eqnarray*}

We do not actually know the terms on the right side of those
equations, so let us parameterize both models with $\theta$ and
rewrite the equations as:


\begin{multline*}
  p(G,B|m_{ch}) = \\
  \int \left ( \sum_D p(G|\theta) p(D|G,\theta) 
  p(B|D,\theta) \right ) p(\theta) d\theta \\
  p(G,B|m_{co}) = \int p(G|\theta)p(B|\theta)p(\theta) d\theta
\end{multline*}

We can learn $p(\theta)$ from available data using dirichlet priors.
The integrals would be difficult algebraically, but they can be
adequately approximated with monte-carlo sampling.

Once we have these, let $n_{g,b}$ be the count of datapoints with
G=g,B=b and we can use:

\begin{equation*}
p(n_*|m) \propto \prod_{g,b} p(g,b|m)^{n_{g,b}}
\end{equation*}

This is proportional instead of equal because there is a combinatoric term,
but it cancels when taking odds ratios so it can be safely ignored.
From here, we can apply standard bayesian updating.

\subsection{Non-Simply-Connected Graphs}

What if there is a causal effect $G\rightarrow B$?  It must be weak
enough not to be detected, but that doesn't say much.  Let us consider
it by cases.

\subsubsection{False Colliders}

Can a true graph of
$G\tikzmark{a}\rightarrow D \rightarrow{B}\tikzmark{c}$
\underarrow{a}{c}
produce a $p(G,B)$ more similar to a graph
$G \rightarrow D \leftarrow B$?  This would require the direct and
indirect influence of $G$ on $B$ to cancel out rather precisely.  If
the indirect influence is stronger, we will pick the correct model,
albeit underconfidently.  If the direct influence is too strong, we
will see a correlation between $G$ and $B$.  And if the direct
influence is in the same direction as the indirect, the correct model
will remain the better fit (though both models will be worse).

\subsubsection{False Chains}

Similarly, can a true graph of
$G\tikzmark{a2}\rightarrow D \leftarrow{B}\tikzmark{c2}$
\underarrow{a2}{c2}
produce a $p(G,B)$ more similar to a graph
$G \rightarrow D \rightarrow B$?  Again, this is possible, but there
is no reason for $p(B|G)$ to resemble $\sum_D p(B|D)p(D|G)$.  If
$p(B|G)-p(B|\bar{G})$ is too large, $G\not\indep B$ will be detected
initially, whereas if it's too small, the correct model will be
chosen.  Furthermore, what shrinks the upper bound is for the
predicted $G-B$ relation in the chain model to be small, which also
means any erroneous bayes factor will be small.

\subsection{Unobserved Confounders}

This technique is vulnerable to confounders.  Specifically, if the
$D-B$ link is the product of a common cause, that will generate no
$G-B$ dependence, exactly like a collider.  That is, it will suggest
we ought to use that bacterium as a treatment, though in fact doing so
would be ineffective.

\section{Severing}

Suppose we do not have a known causal effect to expand from.  The
standard oracle-requiring algorithms do not need one.  It is probably
impossible for a formula to say that $A\not\indep B$ in full
generality, but it may be possible to say $A\not\indep B|S$
sufficiently for our purposes.

Suppose $A$, $B$ and $S$ are variables already known to correlate.  The simplest
associated models are $S-A-B$, $A-S-B$ and $A-B-S$, where $S$ severs only in
the $A-S-B$ case (the directions on the arrows are not important at the
moment, except that there can be no colliders).  For the first two
models:

\begin{eqnarray*}
  p(A,B|S,m_{\cancel{sever}}) & = & p(A|S)p(B|A) \\
  p(A,B|S,m_{sever}) & = & p(A|S)p(B|S)
\end{eqnarray*}

And there is a symmetric formula for the last one.  Since there are
two alternative hypotheses, we take whichever produces the greater
probability.  Somewhat surprisingly, there is no advantage here in
using monte-carlo posteriors instead of simple maximum likelihood estimators
for the conditional probabilities.

There is another simple causal graph that can cause three variables to
correlate: all could be influenced by a common confounder $H$ (for
``hidden'').  This case is impossible to test for in full generality,
because $S$ could follow $H$ so closely as to be an acceptable proxy,
and therefore controlling for $S$ effectively controls for $H$.  In
theory, it should be possible to infer $H$ and its conditionals usings
a Gibbs-sampler-like system.  In practice, there is more room to
overfit this more complex model and no straightforward way to
compensate.  What does work surprisingly well is to ignore this case
and apply the exact same test as before.

\subsection{Applying Severing}

If we are willing to trust the severing test, can we use it to find
causes of a variable of interest?  Deducing the entire causal graph
is a brittle endevour, but there is a local solution.

Suppose there exist three variable (bacteria) with a common (hidden) cause,
such that one causes the variable of interest (disease), that is
$H\tikzmark{h} \rightarrow B_1 \rightarrow D \; B_2\tikzmark{b3} \;
B_3\tikzmark{c3}$ \underarrow{h}{b3} \underarrow{h}{c3}.  The trio
$B_{1,2,3}$ can be identified because they
all correlate and remain correlated when controlling for any of them.
No special test is needed for this, standard $\chi^2$ can produce a p
value for all of this.  The role of $B_1$ can be identified because it
severs $B_2$ and $B_3$ from $D$.  No other simply connected graph has
these properties.

Note that this does not attempt to find \textit{all} causes of $D$,
only those for which suitable $H$, $B_2$ and $B_3$ exist.

\subsection{Unobserved Confounders}

This technique is robust against unobserved confounders.  If the links
between $H$ and $B_{1-3}$ in the preceding graph are
actually mutual effect from a confounder, then that confounder can be
thought of as a part of $H$ (which is itself unobserved).  If the link
between $A$ and $D$ is via confounder, then there will be no $B-D$ or
$C-D$ link to sever.

\subsection{Non-Simply-Connected Graphs}

These techniques depend on simply-connected graphs.  Both the severing
test and its application can say nothing of significance in the face
of multiple connections.

\section{Parsimony}

Given that it will be necessary to assume either ``no unobserved
confounders'' or ``simple connection'', can we at least say that these
are the most parsimonious explanations for our data?

From a graphical model perspective, they are not.  It is the absence
of a link which is a statement about the world.

From a biochemical perspective, they are.  Each link asserts an
interaction, and the default for organic molecules or micro-organisms
is not to interact with each other.  As for unobserved confounders, the
presence of unobserved variables in biology is a given, but for them
to be confounders requires \textit{two} interactions that relate to
each other.

Fields other than biology will need to reconsider this.

\section{Testing in Simulation}

\subsection{Graph Extending}

Empirically, this
works well.  With all probabilities chosen at random, the mean log
probability it assigns to the wrong model is only -0.62.  Furthermore,
if the output is bucketed by posterior, the fraction correct forms a
graph very similar to a straight line (see figure \ref{dir_pla58}).
It does show a minor bias toward chain models, but it goes away at the
edges, and collider models are the useful ones, so this is a safe bias.

While the test is well-calibrated, it returns bayes factors very close
to one most of the time.  For random probabilities and 58 datapoints,
it only gives a clear answer (i.e. a bayes factor outside the range
[0.1,10] 8\% of the time).  This includes cases in which one of the
causal effects is very weak.  Altering the generative function to more
closely resemble the Crohn's data (by having the real p(NOD2) and
p(Crohn's$|$NOD2) and detectable $B\not\indep C$) brings this up to
13\%.  Even with realistic data, 500 datapoints are needed to gain
conclusive answers more than half the time (see figure
\ref{dir_use}).  The realistic data does make the test somewhat less
calibrated (figure \ref{dir_crohns58}), which probably reflects a
collide model being more likely to produce Crohns-like data in the
first place, but the errors remain small and on the side of safety.

\subsubsection{Non-Simply-Connected Graphs}

These examinations are difficult to make rigorous, but simulation
suggests there is little to worry about.  If an $A\rightarrow C$ link
is added but bounded by a $\chi^2$ test, there is a tendency toward
underconfidence, but little other effect (see figure
\ref{dir_multi58}.  This applies to the Crohns-like case, and does not
fully generalize.

Overall, this technique is not \textit{brittle} in the face of an
$A \rightarrow C$ link, but it is not robust against an arbitrarily
strong one either.  Domain knowledge or other tools must be used to
establish safety.  In the case of Crohn's disease, the generally
understood role of NOD2 as mediating \textit{untargetted} immune
responses, makes an $A\rightarrow C$ link unlikely [\cite{nod2}].

\subsection{Severing}

Empirically, severing vs. chain has mean log p(false) of -.41 and
vs. common cause has of -.49.  Both follow close to a straight line
(see figure \ref{sever}).

The applied severing technique does not try to find \textit{all}
causes of $D$, only
those which happen to fit this motif and are strong enough for a
$\chi^2$ test to pick up.  As such, it is unclear what it would mean
to test it in simulation.  Certainly with parameters chosen for the
purpose, it will work.  With random parameters, it will generally fail
to find trios.

\section{Crohn's Disease}

Returning to the original motivating problem, what do these techniques
show for Crohn's Disease?

The bacterial concentrations can be converted to boolean values by
placing thresholds on them.  For each species, a threshold was chosen that
maximized the number of Crohn's Disease statuses that could be
correctly predicted by that bacterium alone.  These boolean values can
then be thought of as ``has a clinically relevant dosage of the
bacterium''.  Once the values were booleanized, all those with
entropies less than 0.5 were dropped on the logic that a species
that's always the same can't have a clinical effect.  This left 222
species.

Of these the direction technique found 8 which show signs of impacting
Crohn's Disease, though only one (Lactobacillus acidophilus) has a
bayes factor greater than 10.  The 8 are shown in figure \ref{dir_tab}.

Inconveniently, that species appears to
be \textit{harmful}, which is surprising given the usual role of
lactobacilli but less so given that two RCTs found this same result
(albeit nonsignificantly) [\cite{lgg1,lgg2}].  It may be relevant that
L. acidophilus is usually found in the
\textit{jejunum} [\cite{lacid}], but these samples are from
\textit{ileum} biopsies.  Perhaps the cause of Crohn's Disease is not
which bacteria are in the intestine, but where in the intestine they
are.  This result is inconvenient because removing or moving bacteria
is far more difficult than adding them.

The severing test offers more possible species (see figure
\ref{sev_tab}).  The bayes factors are higher, but the overall
complexity of the test is reason to be nervous.  This test also took
extremely long to run, which may require fixing before it can be used
seriously.

The species without strong bayes factors are still encouraging.  The
p-value confirms they relate to Crohn's Disease in some way, in which
case ``causes'' and ``is caused by'' are the most likely models by
simple priors.  These small values bump our posterior belief some.
This is not enough to recommend to patients, but it is a better place
to start RCTs than anything we had before.

\newpage

\printbibliography

\newpage
\onecolumn
\section{Figures}

\begin{figure}[h!]
  \includegraphics[width=\textwidth]{nonnormal}
  \caption{Histogram of representative interesting bacterial species.
    The fractions are rounded up before taking logs to avoid log(0).
    These could be approximated as normal, albeit not especially well,
    were it not for the giant spikes on the left.  Perhaps a compound
    model would be possible, but so many degrees of freedom become
    tricky to work with.}
  \label{nonnormal}
\end{figure}
  
\begin{figure}
  \includegraphics[width=\textwidth]{direction_platonic_58}
  \caption{Simulated data using 10k each chain and collide models with
    flatly random parameters, using 0.05-sized buckets for the
    posterior.  Higher numbers indicate collider.}
  \label{dir_pla58}
\end{figure}

\begin{figure}
  \includegraphics[width=\textwidth]{usefullness}
  \caption{Fraction of bayes factors that were at least 10:1 one way
    or the other as a function of dataset size}
  \label{dir_use}
\end{figure}

\begin{figure}
  \includegraphics[width=\textwidth]{direction_crohns_58}
  \caption{The same simulation as \ref{dir_pla58}, but with p(NOD2) and
    p(ICD$|$NOD2) taken from data, and the independences measurable by
    $\chi^2$ test.}
  \label{dir_crohns58}
\end{figure}

\begin{figure}
  \includegraphics[width=\textwidth]{direction_multi_58}
  \caption{The same simulation as \ref{dir_crohns58}, but with
    allowing an $A\rightarrow C$ causal link, albeit one which does
    not show up as $A\not\indep C$ on a $\chi^2$ test.}
  \label{dir_multi58}
\end{figure}

\begin{figure}
  \includegraphics[width=\textwidth]{sever}
  \caption{Calibration for the severing test, both in the case it was
    designed for and in the case of comparing severing to common
    cause.  Higher probabilities indicate severing.}
  \label{sever}
\end{figure}

\begin{figure}
  \begin{tabular}{llrr}
    Species & Sick When & P-Value ICD link & Bayes Factor Causality \\
    Streptococcus pseudopneumoniae & $>$6.36E-05 & 3.1e-05 & 5.15 \\
    Streptococcus infantis & present & 0.0014 & 2.55 \\
    Lactobacillus acidophilus & $>$8.15E-05 & 0.00017 & 10.64 \\
    Sphingopyxis alaskensis & present & 0.0004 & 2.43 \\
    Clostridium methylpentosum & $\leq$6.31E-04 & 0.00085 & 2.53 \\
    Roseiflexus castenholzii & present & 7.3e-05 & 3.30 \\
    Ruminococcus faecis & $\leq$1.07E-03 & 0.0022 & 2.40 \\
  \end{tabular}
  \caption{Results of the direction test on 222 interesting species,
    using $\chi^2$ to check a relationship to Crohn's Disease and the
    direction test described here to establish that it causes (or
    prevents) the disease.}
  \label{dir_tab}
\end{figure}

\begin{figure}
  \begin{tabular}{llllr}
    Species & Sick When & Severs & Third in Trio & Bayes Factor \\
    Coprococcus catus & $\leq$2.67E-04 & B intestinihominis & O splanchnicus & 33.34 \\
Faecalibacterium prausnitzii & $\leq$1.43E-02 & B xylanolyticus & P hordei & 28.48 \\
Flavobacterium thermophilum & $>$2.23E-04 & S infantis & A flavithermus & 83.28 \\
Clostridium glycyrrhizinilyticum & $\leq$4.84E-04 & R faecis & A johnsonii & 46.54 \\
Eubacterium oxidoreducens & $\leq$1.57E-03 & C eutactus & C cellobioparum & 32.67 \\
Bacteroides helcogenes & $\leq$5.10E-04 & L crispatus & B helcogenes & 18.03 \\
Anoxybacillus flavithermus & $>$2.97E-04 & G subterraneus & F thermophilum & 337.11 \\
Clostridium formicaceticum & $\leq$1.74E-04 & A indistinctus & F prausnitzii & 13.29 \\
Streptococcus oralis & $>$1.07E-04 & S sanguinis & V dispar & 13.49 \\
Sporobacter termitidis & $\leq$2.47E-04 & E ruminantium & T forsythia & 106.95 \\
  \end{tabular}
  \caption{Incomplete output of the trio-and-sever test.  All trios
    established p<0.01.  The first species is the one that appears to
    impact Crohn's Disease; the second is the one that the correlates
    with disease until the first is controlled for, and the third is a
    third with a common cause.}
  \label{sev_tab}
\end{figure}

\end{document}
